examples/Makefile.am

```diff
@@ -19,14 +19,44 @@ AM_CFLAGS = -I$(top_builddir)/include -I$(top_srcdir)/include \
 
 #
 
+if ENABLE_CLIENT_SUPPORT
```

Comment:
This is really hard to parse. I suggest using a temporary variable to construct the contents of `bin_PROGRAMS` etc., apply `foreach ....@LIBCOAP_DTLS_LIB_EXTENSION_NAME@` according to `BUILD_ADD_DEFAULT_NAMES`, and then set `bin_PROGRAMS`, `noinst_PROGRAMS` etc. to the result.

Reply:
Reworked Makefile.am using something I learnt recently about automake makefiles - it should now be considerably easier to parse.

Code changes pushed.